src/Acuminator/Acuminator.Vsix/Utils/Version/VSVersionProvider.cs

```diff
@@ -36,27 +36,68 @@ public static async Task<VSVersion> GetVersionAsync(IAsyncServiceProvider servic
             if (vsAppId != null &&
                 vsAppId.GetProperty((int)VSAPropID.VSAPROPID_ProductDisplayVersion, out object? versionObj) == VSConstants.S_OK && versionObj != null)
             {
-                var vsAppIdVersion = new Version(versionObj.ToString());
+                var vsAppIdVersion = new Version(ParseVersion(versionObj));
                 return new VSVersion(vsAppIdVersion);
             }
 
             Version? fileVersion = GetFileVersion();
 
             if (fileVersion != null)
                 return new VSVersion(fileVersion);
-			
+
             var dte = await serviceProvider.GetServiceAsync<EnvDTE.DTE>();
 
             if (!string.IsNullOrWhiteSpace(dte?.Version))
-			{
+            {
                 Version dteVersion = new Version(dte?.Version);
                 return new VSVersion(dteVersion);
-			}
+            }
 
             var unknownVersion = new Version(VSVersion.UnknownVersion, VSVersion.UnknownVersion);
             return new VSVersion(unknownVersion);
         }
 
+        private static string ParseVersion(object? versionObj)
+        {
+            string versionString = versionObj?.ToString() ?? "";
+
+            if (versionString.Contains("Preview"))
+            {
+                /*
```

Comment:
Good comment. But it would be even more helpful if it was done as XML comment on the method, extra details may go into the <remark> section.